static/js/brand-store/store/index.ts

```diff
@@ -14,3 +14,5 @@ export default configureStore({
     invites: invitesSelector,
   },
 });
+
+export type AppDispatch = typeof store.dispatch;
```

Comment:
Have you tried to infer the `RootState` from the store?
```tsx
export type RootState = ReturnType<typeof store.getState>
```

Reply:
Yes, it causes problems with the `Settings` component. Something to come back to.